metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/client/ElasticsearchClient.java

```diff
@@ -0,0 +1,147 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.elasticsearch.client;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.entity.StringEntity;
+import org.apache.metron.common.utils.JSONUtils;
+import org.apache.metron.elasticsearch.utils.FieldMapping;
+import org.apache.metron.elasticsearch.utils.FieldProperties;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+
+/**
+ * Wrapper around the Elasticsearch REST clients. Exposes capabilities of the low and high-level clients.
+ */
+public class ElasticsearchClient implements AutoCloseable{
+  private RestClient lowLevelClient;
+  private RestHighLevelClient highLevelClient;
+
+  public ElasticsearchClient(RestClient lowLevelClient, RestHighLevelClient highLevelClient) {
+    this.lowLevelClient = lowLevelClient;
+    this.highLevelClient = highLevelClient;
+  }
+
+  public RestClient getLowLevelClient() {
+    return lowLevelClient;
+  }
+
+  public RestHighLevelClient getHighLevelClient() {
+    return highLevelClient;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if(lowLevelClient != null) {
+      lowLevelClient.close();
+    }
+  }
+
+  public void putMapping(String index, String type, String source) throws IOException {
+    HttpEntity entity = new StringEntity(source);
+    Response response = lowLevelClient.performRequest("PUT"
+            , "/" + index + "/_mapping/" + type
+            , Collections.emptyMap()
+            , entity
+    );
+
+    if(response.getStatusLine().getStatusCode() != 200) {
+      String responseStr = IOUtils.toString(response.getEntity().getContent());
+      throw new IllegalStateException("Got a " + response.getStatusLine().getStatusCode() + " due to " + responseStr);
+    }
+  }
+
+  public String[] getIndices() throws IOException {
```

Comment:
Are there unit tests specifically for this class?  I don't see any, but maybe I missed.

Reply:
Not unit tests, but the integration tests definitely cover this bit.